pcdsdevices/rtds_ebd.py

```diff
@@ -44,6 +46,27 @@ def _do_move(self, state):
             self.out_cmd.put(1)
 
 
+class RTDSX0ThreeStage(BaseInterface, LightpathMixin):
+    """Three stages X,Y,Z, for solid drilling experiments"""
+
+    mmsx = Cpt(BeckhoffAxis, ':MMS:X', kind='normal')
+    mmsy = Cpt(BeckhoffAxis, ':MMS:Y', kind='normal')
+    mmsz = Cpt(BeckhoffAxis, ':MMS:Z', kind='normal')
+
+    # Determine if the y stage is at it's outer limit switch
+    open_limit = Cpt(EpicsSignalRO, ':MMS:Y.HLS', kind='normal',
+                     doc='Reads if the y-stage is at its outer limit.')
+    lightpath_cpts = ['open_limit']
+
+    def calc_lightpath_state(self, open_limit=None):
+        # Logic, calculations using open_limit for y-stage
+        status = LightpathState(
+            inserted=not open_limit, removed=open_limit,
+            output={'K0' : 0.0}
```

Comment:
Not fully used to the newest lightpath interface yet: does this mean it _always_ shows as blocking the `K0` line in `lightpath`, regardless of inserted or removed?

Does the hard-coding of `K0` here mean that we need to change this when this gets moved to the `L0` line later? Is there a way to not hard-code this?

Reply:
tbh I'm not entirely sure, this is more-or-less following a standard example and some additional assistance from @tangkong. I would appreciate to hear the answer to this as well if you get a moment Robert!

Reply:
(sorry for the tardiness)

Zach is correct, this would show as 0 transmission (blocking beam) no matter the insertion state.  this would also need to be changed if we moved the device as Zach notes.  I'd suggest something like
```python
trans = 0.0 if not open_limit else 1.0
status = LightpathState(
    inserted = not open_limit,
    removed = open_limit,
    output = {self.output_branches[0]: trans}
)
```

if there's additional nuance to the transmission here, we you could code it in prior to creating the `LightpathState`

Reply:
In the end I guess we only check the transmission if the device is inserted, but having the LightpathState return sensible transmissions is easy to do and simple enough I think.

I assume that because of this, lightpath behaved as expected while you were testing?